src/platform/mbed/net/src/net_if.cpp

```diff
@@ -101,14 +103,14 @@ unsigned int mbed_if_nametoindex(const char * ifname)
     if (ifname == NULL)
     {
         errno = ENXIO;
-        return NULL;
+        return 0;
```

Comment:
In C++ it is better to use `nullptr` instead of the int literal `0` (whether it is through `NULL`). `0` doesn't play well with automatic type deduction: 

```c++
auto nil = nullptr;    // converted into a nullptr_t
void* p = nil;           // OK, a nullptr_t can be converted into any ptr!

auto NIL = NULL;   // The compiler deduce NIL is an int because NULL is equal to `0`
void* p2 = NIL;       // Error, the NIL is an int.

```

Reply:
Right, I will change this

Reply:
Sorry but I noticed that this function return unsigned int. In this place, it is correct to return 0.